### PR TITLE
Update to plonk 0.8.2

### PR DESCRIPTION
**Pull request by MiguelDD1**
# Changes

- [x] dusk-plonk updated to 0.8.2 as a dependency. We are using the tag `v0.8.2-polygon-1`.
- [x] Fix dependency issues with rand & libp2p (snow conflict).
- [x] Fix indentation & UTs.
- [x] Fix `crates.io.path`

# Key files to review
- `Cargo.toml` uses `patch` section to override the `plonk` dep.
- Crate `kate`
   - It uses `allocate` feature to integrate with `plonk` on `std`.
   - New `testnet::public_params` fn to set the public params up from a seed.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -194,6 +194,9 @@ members = [
 	"utils/wasm-builder",
 ]
 
+[patch.crates-io]
+dusk-plonk = { git = "https://github.com/maticnetwork/plonk.git", tag = "v0.8.2-polygon-1" }
```
Comment (prabal-banerjee) [inline Cargo.toml:198]: What is the key change in this fork as compared to the original dusk-plonk?

File: Cargo.toml [hunk 2]
```diff
@@ -210,8 +213,6 @@ members = [
 #
 # This list is ordered alphabetically.
 [profile.dev.package]
-aes-soft = { opt-level = 3 }
-aesni = { opt-level = 3 }
```
Comment (prabal-banerjee) [inline Cargo.toml:214]: It looks like we are removing the use of the instruction set, which means slowdown in performance. Is this deliberate or was this part of upstream changes?

File: bin/node-template/node/Cargo.toml [hunk 3]
```diff
@@ -51,12 +51,13 @@ sc-network = { version = "0.9.0", path = "../../../client/network" }
 sc-sync-state-rpc = { version = "0.9.0", path = "../../../client/sync-state-rpc" }
 sp-keystore = { version = "0.9.0", path = "../../../primitives/keystore" }
 
-dusk-plonk = { path = "../../../client/plonk" }
+# dusk-plonk = { path = "../../../client/plonk" }
+dusk-plonk = { version = "0.8.2", default-features = false }
```
Comment (prabal-banerjee) [inline bin/node-template/node/Cargo.toml:55]: In which cases are we using our fork and in which cases are we using the original dusk-plonk implementation? Can we make it uniform across the project?

File: client/kate/Cargo.toml [hunk 4]
```diff
@@ -7,30 +7,38 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-dusk-plonk = { path = "../plonk", default-features = false, optional = true }
+dusk-plonk = { version = "0.8.2", default-features = false, optional = true }
```
Comment (prabal-banerjee) [inline client/kate/Cargo.toml:10]: Earlier we used local plonk code, now using dusk-plonk code. Same question as above - when are we using which?

**Review by prabal-banerjee — APPROVED** [review]
Minor comments to improve my understanding about the changes. Looks good to be merged.

File: Cargo.toml (hunk 1 above)
Comment (MiguelDD1) [reply Cargo.toml:198]: Basically these changes are just to **make "public" some functions, types, and members** we are using in our code. You can see the diff between `0.8.2` and `polygon/0.8.2` [here](https://github.com/maticnetwork/plonk/compare/v0.8.2...polygon/0.8.2)

File: Cargo.toml (hunk 2 above)
Comment (MiguelDD1) [reply Cargo.toml:214]: They are **not** used in this version. Hopefully latest versions of compiler warn about that:
```
❯ cargo b --release
warning: profile package spec `aes-soft` in profile `dev` did not match any packages
warning: profile package spec `aesni` in profile `dev` did not match any packages

        Did you mean `aes`?
   Compiling proc-macro2 v1.0.32
   Compiling syn v1.0.81
   Compiling libc v0.2.107
   Compiling memchr v2.4.1
   Compiling typenum v1.14.0
   Compiling futures-core v0.3.17
   Compiling futures-channel v0.3.17
   Compiling anyhow v1.0.45
   Compiling getrandom v0.1.16
```

File: bin/node-template/node/Cargo.toml (hunk 3 above)
Comment (MiguelDD1) [reply bin/node-template/node/Cargo.toml:55]: We are using `[patch]` section in the `cargo.toml` of the `workspace` . That allows us to **override the `dusk-plonk` version in any crate** built on this `workspace`. 

More info about [Overriding dependencies](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html) and our use-case [working with an unpublished minor version](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#working-with-an-unpublished-minor-version)

File: client/kate/Cargo.toml (hunk 4 above)
Comment (MiguelDD1) [reply client/kate/Cargo.toml:10]: We are using our overridden version, see [above comment](https://github.com/maticnetwork/substrate/pull/6#discussion_r755191982).
You can also verify that the resolved version is right in [Cargo.lock:1502](https://github.com/maticnetwork/substrate/blob/4484f6c13dfe25a9eea79b1c29638dfcdcf0a037/Cargo.lock#L1502)

**Comment by prabal-banerjee** [comment]
Changes look good and nothing breaking. Let's go ahead with the merge.

**Pull request merged**